pandas/core/series.py

```diff
@@ -2461,6 +2461,10 @@ def asof(self, where):
 
         If there is no good value, NaN is returned.
 
+        Note that this is really just a convenient shorthand for `Series.reindex`,
+        and is equivalent to `s.dropna().reindex(where, method='ffill')` for
+        an array of dates.
```

Comment:
The doc string to me means something like: When it's an array of dates, that long formula is an equivalent way of doing it.  I'm not familiar with the issue itself tho.